### PR TITLE
SCICD-533: Resume and Restart Functionality

### PR DESCRIPTION
**Pull request by larry-tordsen**
Add the resume and restart functionality.  Also adds ctrl-c interrupt handling.

## Issues and Related PRs
* Resolves SCICD-533 https://jira-pro.its.hpecorp.net:8443/browse/SCICD-533

### Tested on:
  * frigg

### CODE REVIEW

File: iuf
Comment (joshua-plautz) [inline iuf]: "alogging"

File: iuf
Comment (larry-tordsen) [inline iuf]: I think this is a rebase error.  Get rid of everything except `config.activity.restart(config)`

File: iuf
Comment (deantroe) [inline iuf]:
```suggestion
        print("    Enter Y/y/Yes to abort after the current stage")
```

Comment (deantroe) [reply iuf]: This is to make it consistent with the line below (although capitalized "Yes" is still different that lowercase "background" so I guess we should make that match too.

File: iuf
Comment (deantroe) [reply iuf]: And there are two periods.

File: iuf
Comment (deantroe) [inline iuf]: Did you intend to change this wording?

File: iuf
Comment (deantroe) [inline iuf:585]: Is that all the arguments needed, or do they need to specify any of the other arguments they originally used before they did a suspend?

**Comment by deantroe** [comment]
restart and resume both have ```comment``` arguments, but they are not suppressed nor do they have descriptions.  We should do one of those, depending on whether ```comment``` actually works at the moment.

**Comment by deantroe** [comment]
Here's the help strings for restart, resume, and abort respectively:
```
Restart a previously paused or failed IUF session for a given activity.
Resume a previously paused or failed IUF session for a given activity.
Abort an IUF session for a given activity after the current stage completes.
```

I don't think there's any user-facing description of what a "paused session" means.  Seems like we should use the term "aborted" instead?  And "stage" instead of "session"?

**Comment by deantroe** [comment]
Should we add a sentence to the help output of ```iuf run``` telling the reader that Ctrl-C can be used to abort or background the ```iuf run``` command?

File: iuf
Comment (JohnNavitskyHPE) [inline iuf]: I'd consider describing this as "disconnecting".  
```
Enter D, d, or disconnect to exit the IUF CLI.  The install will continue in the background, however no logs will be collected.
```

File: iuf
Comment (JohnNavitskyHPE) [inline iuf]: You might give them an entry in the above list to resume monitoring, like:
```
Enter <return> to resume monitoring.
```

**Comment by JohnNavitskyHPE** [comment]
> Here's the help strings for restart, resume, and abort respectively:
> 
> ```
> Restart a previously paused or failed IUF session for a given activity.
> Resume a previously paused or failed IUF session for a given activity.
> Abort an IUF session for a given activity after the current stage completes.
> ```
> 
> I don't think there's any user-facing description of what a "paused session" means. Seems like we should use the term "aborted" instead? And "stage" instead of "session"?

Yes,
```
Restart a previously failed or aborted IUF session for a given activity.
```

```
Resume a failed IUF session for a given activity.
```

File: iuf
Comment (JohnNavitskyHPE) [reply iuf:585]: I believe that's all they need.

**Review by JohnNavitskyHPE — APPROVED** [review]

**Review by deantroe — APPROVED** [review]
@larry-tordsen I am approving, but it would be great if the remaining comments (just wording fixups) could be addressed before you merge.

File: lib/InstallerUtils.py
Comment (larry-tordsen) [inline lib/InstallerUtils.py:84]: I was hitting an AttributeError exception on the yaml.safe_load here.  It's in the product catalog as yaml, but it seems to get converted from yaml when the `json.loads` is called.

**Pull request merged**
